app/Traits/PageFetcher.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Traits;
+
+use Illuminate\Support\Facades\Http;
+
+trait PageFetcher
+{
+    private string $apiUrl;
+
+    //this function is used to fetch pages on namespace
+    function fetchPagesInNamespace(string $wikiDomain, int $namespace): array
+    {
+        $this->apiUrl = getenv('PLATFORM_MW_BACKEND_HOST').'/w/api.php';
```

Comment:
Nitpick: I feel the trait reading configuration values all by itself is a bit of "overstepping" its own responsibilities (the trait shouldn't have any idea how it's being configured), so how about:

- setting the `apiUrl` prop in the class constructors as before?
- in this method, raise a `RuntimeException` in case it hasn't been set